tests/models/test_param.py

```diff
@@ -22,278 +22,13 @@
-        assert dump["schema"] == {"type": "string", "minLength": 2}
-
-    @pytest.mark.parametrize(
-        "param",
-        [
-            Param("my value", description="hello", schema={"type": "string"}),
-            Param("my value", description="hello"),
-            Param(None, description=None),
-            Param([True], type="array", items={"type": "boolean"}),
-            Param(),
-        ],
-    )
-    def test_param_serialization(self, param: Param):
-        """
-        Test to make sure that native Param objects can be correctly serialized
-        """
-
-        serializer = BaseSerialization()
-        serialized_param = serializer.serialize(param)
-        restored_param: Param = serializer.deserialize(serialized_param)
-
-        assert restored_param.value == param.value
-        assert isinstance(restored_param, Param)
-        assert restored_param.description == param.description
-        assert restored_param.schema == param.schema
-
-
-class TestParamsDict:
-    def test_params_dict(self):
-        # Init with a simple dictionary
-        pd = ParamsDict(dict_obj={"key": "value"})
-        assert isinstance(pd.get_param("key"), Param)
-        assert pd["key"] == "value"
-        assert pd.suppress_exception is False
-
-        # Init with a dict which contains Param objects
-        pd2 = ParamsDict({"key": Param("value", type="string")}, suppress_exception=True)
-        assert isinstance(pd2.get_param("key"), Param)
-        assert pd2["key"] == "value"
-        assert pd2.suppress_exception is True
-
-        # Init with another object of another ParamsDict
-        pd3 = ParamsDict(pd2)
-        assert isinstance(pd3.get_param("key"), Param)
-        assert pd3["key"] == "value"
-        assert pd3.suppress_exception is False  # as it's not a deepcopy of pd2
-
-        # Dump the ParamsDict
-        assert pd.dump() == {"key": "value"}
-        assert pd2.dump() == {"key": "value"}
-        assert pd3.dump() == {"key": "value"}
-
-        # Validate the ParamsDict
-        plain_dict = pd.validate()
-        assert isinstance(plain_dict, dict)
-        pd2.validate()
-        pd3.validate()
-
-        # Update the ParamsDict
-        with pytest.raises(ParamValidationError, match=r"Invalid input for param key: 1 is not"):
-            pd3["key"] = 1
-
-        # Should not raise an error as suppress_exception is True
-        pd2["key"] = 1
-        pd2.validate()
-
-    def test_update(self):
-        pd = ParamsDict({"key": Param("value", type="string")})
-
-        pd.update({"key": "a"})
-        internal_value = pd.get_param("key")
-        assert isinstance(internal_value, Param)
-        with pytest.raises(ParamValidationError, match=r"Invalid input for param key: 1 is not"):
-            pd.update({"key": 1})
-
-    def test_repr(self):
-        pd = ParamsDict({"key": Param("value", type="string")})
-        assert repr(pd) == "{'key': 'value'}"
-
-
 class TestDagParamRuntime:
```

Comment:
I think we might need to move these too?

This is testing the runtime resolution behavoiur (i.e. that the task is called with the value, not the Param object, and that dagrun.conf is used in preference over the dag params.